api/client/hijack.go

```diff
@@ -196,19 +193,15 @@ func (cli *DockerCli) hijack(method, path string, setRawTerminal bool, in io.Rea
 		defer term.RestoreTerminal(cli.inFd, oldState)
 	}
 
+	receiveStdout := make(chan error)
```

Comment:
I don't think there is an issue with unbuffered here. It still makes sure to read off whatever is being sent, and only one thing is sending.